---
title: "Single-cell based analysis of Tha astrocyte subpopulations"
author: "Arpy"
date: '2024-09-24'
output: html_document
---

```{r libraries and functions, message=FALSE}
library(tidyverse)
library(ggplot2)
library(dplyr)
library(Seurat)
library(Matrix)
library(viridis)
#library(randomForest)
#library(ranger)


source("~/OHSU Dropbox/Saunders Lab's shared workspace/arpy/manuscripts/2023_Thai2P4M_FeigeYoung/ms_analyses/1_karl_analysis/r_functions_paths/_color_palettes_tha2P4M.R")
source("~/OHSU Dropbox/Saunders Lab's shared workspace/arpy/manuscripts/2023_Thai2P4M_FeigeYoung/ms_analyses/1_karl_analysis/r_functions_paths/_karl_tha2P4M_functions.R")
source("~/OHSU Dropbox/Saunders Lab's shared workspace/arpy/manuscripts/2023_Thai2P4M_FeigeYoung/ms_analyses/1_karl_analysis/r_functions_paths/_karl_tha2P4M_filepaths.R")
```

#0. Load Data
```{r load data, echo=FALSE}

tha.celltype.infections.summary_astrocytes <- readRDS(file = tha.celltype.infections.summary_astrocytes.path)
astro.seurat                               <- readRDS(file = astro.seurat.path)
astro.seurat.markers                       <- readRDS(file = astro.seurat.markers.path)

tha.celltype.infections.summary                 <- read_rds(tha.celltype.infections.summary.path)
tha.celltype.infections.summary_doubletsflagged <- read_rds(tha.celltype.infections.summary_doubletsflagged.path)

tha.seurat                                      <- readRDS(file = tha.seurat.path)

#Key DE results - all CoarseCellTypes
DE.tha5.vs.NI.CoarseCellType 					        <- readRDS(DE.tha5.vs.NI.CoarseCellType.path)
DE.tha05.vs.NI.CoarseCellType 					      <- readRDS(DE.tha05.vs.NI.CoarseCellType.path)

#ISG Database and IFN Gene Sets
isg_database             <- read.csv(file = isg_database.path, header = T, sep = ",")
IFNGenesAll_913531       <- read.csv(file = IFNGenesAll_913531.path,   header = T, sep = "\t")
IFNAlphaBeta_909733      <- read.csv(file = IFNAlphaBeta_909733.path,  header = T, sep = "\t")
IFNGammaGenes_877300     <- read.csv(file = IFNGammaGenes_877300.path, header = T, sep = "\t")

```

#1. Explore Astrocyte Subclusters
```{r explore astro subclusters}

astro.seurat.markers_cluster0 <-
  astro.seurat.markers %>%
  filter(cluster == 0)

astro.seurat.markers_cluster1 <-
  astro.seurat.markers %>%
  filter(cluster == 1)

astro.seurat.markers_cluster2 <-
  astro.seurat.markers %>%
  filter(cluster == 2)

astro.seurat.markers_cluster3 <-
  astro.seurat.markers %>%
  filter(cluster == 3)

astro.seurat.markers_cluster4 <-
  astro.seurat.markers %>%
  filter(cluster == 4)

```

#2. Preprocessing steps
```{r adjust factor levels, echo = FALSE}
#Remove Doublets; save as CLEAN2
tha.celltype.infections.summary_astrocytes_CLEAN <-
  tha.celltype.infections.summary_astrocytes %>%
  mutate(Experiment = factor(Experiment, c("NI_0", "Tha_05", "Tha_5", "2P4M_05"))) %>%
  mutate(astro.include = ident %in% c(0,2) &
    ((2.5*astro_UMAP_1 + astro_UMAP_2) < 10) &
    (astro_UMAP_2 > -8) & (astro_UMAP_2 < 5)
  )

tha.celltype.infections.summary_astrocytes_CLEAN2 <-
  tha.celltype.infections.summary_astrocytes_CLEAN %>%
  filter(astro.include) %>%
  as_tibble()


#Add IFN and InnateImmuneDB scores
###PreProcess 

###0. Gene Lists
#InnateImmuneDB
innatedb_curated_genes <- as_tibble(isg_database)

innatedb_curated_genes.Human <-
  innatedb_curated_genes %>%
  filter(Species == 9606) %>%
  pull(Gene.Symbol) %>%
  as.character() %>%
  unique()

#IFN REACTOME Genes
IFNGenesAll_913531.genes   <- head(x = sub(".* ", "", IFNGenesAll_913531$MoleculeName), -2)
IFNAlphaBeta_909733.genes  <- sub(".* ", "", IFNAlphaBeta_909733$MoleculeName)
IFNGammaGenes_877300.genes <- sub(".* ", "", IFNGammaGenes_877300$MoleculeName)

###1. Extract Normalized DGE
astro.seurat_NormDGE             <- GetAssayData(astro.seurat, slot = "data")

###2. Calculate Normalized Gene Expression Scores
#InnateImmuneDB
ISGs_to_filter                   <- intersect(x = innatedb_curated_genes.Human, y = rownames(astro.seurat_NormDGE))
astro.seurat_NormDGE.ISGs        <- astro.seurat_NormDGE[ISGs_to_filter, , drop = FALSE] #824 ISG genes in the astrocyte DGE
astro.seurat_TotalISGsExpression <- as.data.frame(colSums(astro.seurat_NormDGE.ISGs))
astro.seurat_TotalISGsExpression <- as_tibble(astro.seurat_TotalISGsExpression, rownames = "CBC")
astro.seurat_TotalISGsExpression<- astro.seurat_TotalISGsExpression %>% rename(ISGScore = `colSums(astro.seurat_NormDGE.ISGs)`)

tha.celltype.infections.summary_astrocytes_CLEAN2 <- left_join(x = tha.celltype.infections.summary_astrocytes_CLEAN2, y = astro.seurat_TotalISGsExpression, by = "CBC") #join

#IFNGenesAll_913531
IFNGenesAll_to_filter            <- intersect(x = IFNGenesAll_913531.genes, y = rownames(astro.seurat_NormDGE))
astro.seurat_NormDGE.IFNGenesAll <- astro.seurat_NormDGE[IFNGenesAll_to_filter, , drop = FALSE] #824 ISG genes in the astrocyte DGE
astro.seurat_TotalIFNGenesAll    <- as.data.frame(colSums(astro.seurat_NormDGE.IFNGenesAll))
astro.seurat_TotalIFNGenesAll    <- as_tibble(astro.seurat_TotalIFNGenesAll, rownames = "CBC")
astro.seurat_TotalIFNGenesAll    <- astro.seurat_TotalIFNGenesAll %>% rename(IFNGenesAllScore = `colSums(astro.seurat_NormDGE.IFNGenesAll)`)

tha.celltype.infections.summary_astrocytes_CLEAN2 <- left_join(x = tha.celltype.infections.summary_astrocytes_CLEAN2, y = astro.seurat_TotalIFNGenesAll, by = "CBC") #join



#save tha.celltype.infections.summary_astrocytes_CLEAN2
saveRDS(object = tha.celltype.infections.summary_astrocytes_CLEAN2, file = tha.celltype.infections.summary_astrocytes_CLEAN2.path)


```


#3. UMAP Plotting of cell identities and viral load
```{r define plot path, echo=FALSE}

astro.plot.path <- "~/OHSU Dropbox/Saunders Lab's shared workspace/arpy/manuscripts/2023_Thai2P4M_FeigeYoung/ms_analyses/1_karl_analysis/Plots/6_astrocytes/"
```

```{r UMAP plots, echo=FALSE}

umap_point_size = 1.5

UMAP_astrocytes_Idents_by_Experiment <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident %in% c(0,2)) %>%
    ggplot(aes(x = astro_UMAP_1, y = astro_UMAP_2, color = ident))+
    geom_point(size=umap_point_size) +
    scale_color_manual(values = c(Astrocyte_color, AstrocteReactive_color))+
    facet_wrap(~Experiment, ncol = 4)+
    theme_bw() +
    theme(
      legend.position='none',
      axis.ticks=element_blank(),
      axis.text=element_blank(),
      axis.title=element_blank(),
      strip.background = element_blank(),
      strip.text.x = element_blank()
      )+
    labs(x=NULL, y=NULL)

ggsave(paste0(astro.plot.path, "umap/UMAP_astrocytes_Idents_by_Experiment.png"), UMAP_astrocytes_Idents_by_Experiment, device='png', width = 28, height=8, units='cm', dpi=600)
    

UMAP_astrocytes_Idents_by_ViralLoad <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident %in% c(0,2)) %>%
    filter(ViralLoad > 0) %>%
    ggplot(aes(x = astro_UMAP_1, y = astro_UMAP_2, color = ViralLoad, alpha = ViralLoad))+
    geom_point(size=umap_point_size) +
    scale_alpha_continuous(range = c(0.3, 1))+
    #scale_color_viridis_c(option = "inferno", direction = -1)+
    scale_color_viridis_c(option = "viridis", direction = -1)+
    facet_wrap(~Experiment, ncol = 4)+
    theme_dark() +
    theme(
      #panel.grid.major = element_blank(),
      #panel.grid.minor = element_blank(),
      #panel.background = element_rect(fill = '#525252'),
      legend.position='none',
      axis.ticks=element_blank(),
      axis.text=element_blank(),
      axis.title=element_blank(),
      strip.background = element_blank(),
      strip.text.x = element_blank()
      )+
    labs(x=NULL, y=NULL)

ggsave(paste0(astro.plot.path, "umap/UMAP_astrocytes_Idents_by_ViralLoad.png"), UMAP_astrocytes_Idents_by_ViralLoad, device='png', width = 28, height=8, units='cm', dpi=600)

UMAP_astrocytes_Idents_by_ViralLoad_Legend <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident %in% c(0,2)) %>%
    filter(ViralLoad > 0) %>%
    ggplot(aes(x = astro_UMAP_1, y = astro_UMAP_2, color = ViralLoad, alpha = ViralLoad))+
    geom_point(size=umap_point_size) +
    scale_alpha_continuous(range = c(0.3, 1))+
    #scale_color_viridis_c(option = "inferno", direction = -1)+
    scale_color_viridis_c(option = "viridis", direction = -1)+
    facet_wrap(~Experiment, ncol = 4)+
    theme_dark() +
    theme(panel.background = element_rect(fill = 'black'))+
    labs(x=NULL, y=NULL)

ggsave(paste0(astro.plot.path, "umap/UMAP_astrocytes_Idents_by_ViralLoad_Legend.png"), UMAP_astrocytes_Idents_by_ViralLoad_Legend, device='png', width = 28, height=8, units='cm', dpi=600)

```

#4. Astrocyte Composition
```{r composition bar plots, echo=FALSE}

#Experimental Composition
astrocyte_composition <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  filter(ident %in% c(0,2)) %>%
  group_by(ident) %>%
  count(Experiment)

astrocyte_composition2 <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  count(Experiment) %>%
  mutate(ident = "all") %>%
  relocate(ident)
  
astrocyte_composition_summary <- bind_rows(astrocyte_composition, astrocyte_composition2)

astrocyte_composition_summary <-
  astrocyte_composition_summary %>%
  mutate(ident = factor(ident, levels = c("2", "0", "all"))) %>%
  mutate(Experiment = factor(Experiment, levels = rev(c("2P4M_05", "Tha_5", "Tha_05", "NI_0"))))

astrocyte_composition_barplot <-
  ggplot(astrocyte_composition_summary, aes(x = ident, y = n, fill = forcats::fct_rev(Experiment))) +
   geom_bar(stat = "identity", position="fill", width = .5) +
   #scale_fill_manual(values = rev(c(Tha2P4M_color,  Tha5_color, Tha_color, Uninfected_color))) +
   scale_fill_manual(values = c(Tha2P4M_color,  Tha5_color, Tha_color, Uninfected_color)) +
   coord_flip()+
   theme_bw()+
   theme(aspect.ratio = 1/4)

ggsave(paste0(astro.plot.path, "bar/astrocyte_composition_barplot.png"), astrocyte_composition_barplot, device='png', width = 15, height=8, units='cm', dpi=600)


#Infected/Uninfected Composition
astrocyte_infected_vs_uninfected_composition <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  filter(ident %in% c(0,2)) %>%
  group_by(ident) %>%
  count(infected)

astrocyte_infected_vs_uninfected_composition2 <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  count(infected) %>%
  mutate(ident = "all") %>%
  relocate(ident)
  
astrocyte_infected_vs_uninfected_composition_summary <- bind_rows(astrocyte_infected_vs_uninfected_composition, astrocyte_infected_vs_uninfected_composition2)

astrocyte_infected_vs_uninfected_composition_summary <-
  astrocyte_infected_vs_uninfected_composition_summary %>%
  mutate(ident = factor(ident, levels = c("2", "0", "all"))) %>%
  mutate(infected = factor(infected, levels = c("no", "yes")))


astrocyte_infected_vs_uninfected_composition_barplot <-
  ggplot(astrocyte_infected_vs_uninfected_composition_summary, aes(x = ident, y = n, fill = forcats::fct_rev(infected))) +
   geom_bar(stat = "identity", position="fill", width = .5) +
   scale_fill_manual(values = c(Infected_color, Uninfected_color)) +
   coord_flip()+
   theme_bw()+
   theme(aspect.ratio = 1/4)

ggsave(paste0(astro.plot.path, "bar/astrocyte_infected_vs_uninfected_composition_barplot.png"), astrocyte_infected_vs_uninfected_composition_barplot, device='png', width = 15, height=8, units='cm', dpi=600)


```

#5. Viral Load Density Plots

```{r density plots, echo = FALSE}

#Original histogram bin number
tha.celltype.infections.summary_astrocytes_CLEAN2_log10 <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  filter(ident %in% c(0,2)) %>%
  filter( seurat.include == 'yes',
    infected == 'yes', infected.posterior > 0.8,
    G > 0, L > 0
    )  %>%
  mutate(
    lViralLoad = log10(ViralLoad),
    lTotalHost = log10(TotalHost),
    lTotalViral = log10(TotalViral),
    lTotalUMIs = log10(TotalUMIs),
    ) %>%
  filter(Experiment %in% c('Tha_5', 'Tha_05', '2P4M_05')) 
  
ViralLoad_DensityPlots_log10_Astro.low_vs_Astro.high.plot <- 
  ggplot(tha.celltype.infections.summary_astrocytes_CLEAN2_log10, aes(lViralLoad, y=after_stat(density)))+
    geom_histogram(aes(fill = ident), bins=60, position='identity', alpha=0.6)+
    geom_density(aes(group = ident), color='black', adjust = 1.2, linewidth = 1.7)+
    geom_density(aes(color = ident), adjust = 1.2, linewidth = 1.5)+
    geom_vline(xintercept = -1.5, linetype="dotted", color = "black", linewidth=0.75)+
    scale_color_manual(values = c(Astrocyte_color, AstrocteReactive_color))+
    scale_fill_manual(values =  c(Astrocyte_color, AstrocteReactive_color))+
    scale_x_continuous(name = "% Rabies UMIs", breaks = c(-2.0, -1.5, -1.0, -0.5), labels = c( "1","3", "10", "30"))+
    ylab("Density")+
    theme_bw()+
    theme(plot.margin = ggplot2::margin(t = 0, r = 1, b = 0, l = 1, unit = "cm"),
          axis.text.x = element_text (size = 12,  family = "Arial", face = "plain", color = "black"),
          axis.text.y = element_blank(),
          axis.title = element_text (size =12, family = "Arial", face = "bold", color = "black"),
          strip.background = element_blank(),
          strip.text = element_text (size =16, family = "Arial", face = "bold", color = "black")) 

ggsave(filename =  paste0(astro.plot.path, "density/ViralLoad_DensityPlots_log10_Astro.low_vs_Astro.high.plot.png"), plot = ViralLoad_DensityPlots_log10_Astro.low_vs_Astro.high.plot,
       device='png', width = 18, height=10, units='cm', dpi=600)


#Fewer bins for Reviewer
ViralLoad_DensityPlots_log10_Astro.low_vs_Astro.high.fewerbins.plot <- 
  ggplot(tha.celltype.infections.summary_astrocytes_CLEAN2_log10, aes(lViralLoad, y=after_stat(density)))+
    geom_histogram(aes(fill = ident), bins=15, position='identity', alpha=0.6)+
    geom_density(aes(group = ident), color='black', adjust = 1.2, linewidth = 1.7)+
    geom_density(aes(color = ident), adjust = 1.2, linewidth = 1.5)+
    geom_vline(xintercept = -1.5, linetype="dotted", color = "black", linewidth=0.75)+
    scale_color_manual(values = c(Astrocyte_color, AstrocteReactive_color))+
    scale_fill_manual(values =  c(Astrocyte_color, AstrocteReactive_color))+
    scale_x_continuous(name = "% Rabies UMIs", breaks = c(-2.0, -1.5, -1.0, -0.5), labels = c( "1","3", "10", "30"))+
    ylab("Density")+
    theme_bw()+
    theme(plot.margin = ggplot2::margin(t = 0, r = 1, b = 0, l = 1, unit = "cm"),
          axis.text.x = element_text (size = 12,  family = "Arial", face = "plain", color = "black"),
          axis.text.y = element_blank(),
          axis.title = element_text (size =12, family = "Arial", face = "bold", color = "black"),
          strip.background = element_blank(),
          strip.text = element_text (size =16, family = "Arial", face = "bold", color = "black")) 

ggsave(filename =  paste0(astro.plot.path, "density/ViralLoad_DensityPlots_log10_Astro.low_vs_Astro.high.fewerbins.plot.png"), plot = ViralLoad_DensityPlots_log10_Astro.low_vs_Astro.high.fewerbins.plot,
       device='png', width = 18, height=10, units='cm', dpi=600)

```

#6. Viral Load vs DYNNL1 
```{r DYNNL1 vs Viral Load, echo=FALSE}

#pull the DYNLL1 AND DYNLL2 genes
genes_to_show <- c('DYNLL1', 'DYNLL2')

#use center, scaled log transformed expression data
gene_subset <- 
FetchData(tha.seurat, genes_to_show, slot='data') %>%
  rownames_to_column(var = 'CBC') %>%
  merge(tha.celltype.infections.summary, by = 'CBC')

astro_subset <-
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  select(CBC, astro.include, ident) 

DYNNL1_Summary <- left_join(gene_subset, astro_subset, by = 'CBC')


#preprocess file for plotting
n_gene_tiles <- 3

DYNNL1_Summary_CellTypeStratified <-
  DYNNL1_Summary %>%
  mutate(Neuron_AstroLow_AstroHigh = case_when(ident == "0" ~ "AstroLow",
                                               ident == "2" ~ "AstroHigh",
                                               CoarseCellType == "Neuron" ~ "Neuron",
                                               TRUE ~ NA)) %>%
  mutate(Neuron_AstroLow_AstroHigh = factor(Neuron_AstroLow_AstroHigh, levels = c("Neuron", "AstroLow", "AstroHigh"))) %>%
  mutate(CoarseCellType = factor(CoarseCellType, levels = c("Neuron", "Astrocyte"))) %>%
  filter(Neuron_AstroLow_AstroHigh %in% c("AstroLow", "AstroHigh", "Neuron")) %>%
  filter(infected == "yes") %>%
  arrange(Neuron_AstroLow_AstroHigh) %>%
  filter(ViralLoad > 0.001) %>%
  #filter(CoarseCellType == "Astrocyte") %>%
  #group_by(Neuron_AstroLow_AstroHigh) %>%
  mutate(DYNLL1_groups = as.factor(as.character(ntile(DYNLL1, n_gene_tiles)))) %>% #4
  mutate(ViralLoad_groups = as.factor(as.character(ntile(ViralLoad, n_gene_tiles)))) #4

#plotting
DYNNL1_byViralLoadGroup_byN_vs_Alow_vs_Ahigh.plot <-
      ggplot(DYNNL1_Summary_CellTypeStratified, aes(x = ViralLoad_groups, y = DYNLL1, fill = ViralLoad_groups, group = ViralLoad_groups))+
        geom_jitter(size = 0.75, alpha = .4, color = "white")+
        geom_violin(scale = "width", alpha = 0.8, color = "black")+
        scale_fill_manual(values = rev(viridis_pal(option = "D", begin = 0.7)(n_gene_tiles)))+
        theme_dark()+
        facet_wrap(~Neuron_AstroLow_AstroHigh)


#save plot
ggsave(filename =  paste0(astro.plot.path, "violin/DYNNL1_byViralLoadGroup_byN_vs_Alow_vs_Ahigh.plot.png"), plot = DYNNL1_byViralLoadGroup_byN_vs_Alow_vs_Ahigh.plot, device='png', width = 18, height=6, units='cm', dpi=600)


```


#7. InnateImmuneDB and IFNAll Score Comparison
```{r ISG score, echo = FALSE}

tha.celltype.infections.summary_astrocytes_CLEAN3 <- 
  tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
  mutate(Neuron_AstroLow_AstroHigh = case_when(ident == "0" ~ "AstroLow",
                                               ident == "2" ~ "AstroHigh",
                                               CoarseCellType == "Neuron" ~ "Neuron",
                                               TRUE ~ NA)) %>%
  mutate(Neuron_AstroLow_AstroHigh = factor(Neuron_AstroLow_AstroHigh, levels = c("Neuron", "AstroLow", "AstroHigh"))) %>%
  mutate(CoarseCellType = factor(CoarseCellType, levels = c("Neuron", "Astrocyte"))) %>%
  filter(Neuron_AstroLow_AstroHigh %in% c("AstroLow", "AstroHigh", "Neuron")) %>%
  filter(infected == "yes")
  
#InnateImmuneDB
ISGScore.plot <- 
  ggplot(tha.celltype.infections.summary_astrocytes_CLEAN3, aes(x = Neuron_AstroLow_AstroHigh, y = ISGScore))+
    geom_jitter(size = 0.75, alpha = .6, color = "red")+
    geom_violin(scale = "width", alpha = 0.4, color = "black")+
    theme_dark()

#save plot
ggsave(filename =  paste0(astro.plot.path, "violin/ISGScore.plot.png"), plot = ISGScore.plot, device='png', width = 7, height=6, units='cm', dpi=600)


#InnateImmuneDB
IFNGenesAllScore.plot <- 
  ggplot(tha.celltype.infections.summary_astrocytes_CLEAN3, aes(x = Neuron_AstroLow_AstroHigh, y = IFNGenesAllScore))+
    geom_jitter(size = 0.75, alpha = .6, color = "red")+
    geom_violin(scale = "width", alpha = 0.4, color = "black")+
    theme_dark()

#save plot
ggsave(filename =  paste0(astro.plot.path, "violin/IFNGenesAllScore.plot.png"), plot = IFNGenesAllScore.plot, device='png', width = 7, height=6, units='cm', dpi=600)


```

#8. Pie Charts of Cluster 2 Astrocytes across experimental conditions
```{r cluster 2 (coded as "Cluster 1" here to match the Seurat number percentages across conditions, echo=FALSE}

#0. Make Tibble of the Cluster 2 counts across conditions
#Some count numbers are pulled from "ContingencyTable_All_vs_Cluster2" below
condition <- c("NPH ExVivo", "No Infection", "Tha(0.5)", "Tha(5)", "Tha2P4M(0.5)")
n_Cluster1 <- c(38,    2, 18, 24, 172)
n_Total    <- c(82596, 1072, 1183, 811, 909)

#1. Prepare long-format data with proportions
Cluster2_ProportionsAcrossConditionsSummary <- tibble(
  condition = condition,
  n_Cluster1 = n_Cluster1,
  n_Total = n_Total
) %>%
  mutate(n_Other = n_Total - n_Cluster1) %>%
  pivot_longer(cols = c(n_Cluster1, n_Other),
               names_to = "category", values_to = "count") %>%
  group_by(condition) %>%
  mutate(proportion = count / sum(count)) %>%
  ungroup()

#2. Reorder the condition factor
Cluster2_ProportionsAcrossConditionsSummary$condition <- factor(Cluster2_ProportionsAcrossConditionsSummary$condition, levels = c("NPH ExVivo", "No Infection", "Tha(0.5)", "Tha(5)", "Tha2P4M(0.5)"))

#3. Plot
Cluster2_ProportionsAcrossConditionsSummary_BarPlot <- 
  ggplot(Cluster2_ProportionsAcrossConditionsSummary, aes(x = category, y = proportion, fill = category)) +
    geom_col() +
    facet_wrap(~ condition, nrow = 1) +
    labs(
      title = "Proportion of Cluster 1 vs Other Cells by Condition",
      x = NULL,
      y = "Proportion"
    ) +
    scale_fill_manual(
      values = c("n_Cluster1" = AstrocteReactive_color, "n_Other" = Astrocyte_color),
      labels = c("Cluster 1", "Other")
    ) +
    theme_bw(base_size = 13) +
    theme(
      legend.title = element_blank(),
      strip.text = element_text(face = "bold", size = 16),  # Increased font size here
      axis.text.x = element_text(angle = 45, hjust = 1)
    )

#4. Plot save
#save plot
ggsave(filename =  paste0(astro.plot.path, "bar/Cluster2_ProportionsAcrossConditionsSummary_BarPlot.png"), plot = Cluster2_ProportionsAcrossConditionsSummary_BarPlot, device='png', width = 10, height=3.5, units='in', dpi=600)

```


#9. STATISTICS AND DATA DESCRIPTION
```{r stats and description, echo = FALSE}
#0. Load the cleaned/filterd astrocyte infection summary file
tha.celltype.infections.summary_astrocytes_CLEAN2 <- readRDS(tha.celltype.infections.summary_astrocytes_CLEAN2.path)

#1. Determine the number and percentage of astrocytes in "Cluster 1" (Cluster 0 in Seurat) and "Cluster 2" (Cluster 1 in Seurat)

table(tha.celltype.infections.summary_astrocytes_CLEAN2$ident) #Cluster 1 = 3,759 cells; Cluster 2 = 216 cells
round(table(tha.celltype.infections.summary_astrocytes_CLEAN2$ident) / sum(table(tha.celltype.infections.summary_astrocytes_CLEAN2$ident)), digits = 3)

#2a. Test for enrichment of Tha2P4M cells in Cluster 2 versus the whole dataset
ExperimentalCellCounts_All <- 
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    count(Experiment)

ExperimentalCellCounts_Cluster1 <- 
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident == "0") %>%
    count(Experiment)

ExperimentalCellCounts_Cluster2 <- 
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident == "2") %>%
    count(Experiment) 

#Build Contingency Tables
ContingencyTable_All_vs_Cluster1 <- as.data.frame(matrix(nrow = 2, ncol = 4))
ContingencyTable_All_vs_Cluster2 <- as.data.frame(matrix(nrow = 2, ncol = 4))

colnames(ContingencyTable_All_vs_Cluster1) <- ExperimentalCellCounts_All$Experiment
rownames(ContingencyTable_All_vs_Cluster1) <- c("All", "Cluster1")
ContingencyTable_All_vs_Cluster1[1,] <- ExperimentalCellCounts_All$n
ContingencyTable_All_vs_Cluster1[2,] <- ExperimentalCellCounts_Cluster1$n

colnames(ContingencyTable_All_vs_Cluster2) <- ExperimentalCellCounts_All$Experiment
rownames(ContingencyTable_All_vs_Cluster2) <- c("All", "Cluster1")
ContingencyTable_All_vs_Cluster2[1,] <- ExperimentalCellCounts_All$n
ContingencyTable_All_vs_Cluster2[2,] <- ExperimentalCellCounts_Cluster2$n

#Chi.Sq Testing
ChiSq_All_vs_Cluster1 <- chisq.test(x = ContingencyTable_All_vs_Cluster1, simulate.p.value = F)
ChiSq_All_vs_Cluster2 <- chisq.test(x = ContingencyTable_All_vs_Cluster2, simulate.p.value = F)

ChiSq_All_vs_Cluster1$residuals
ChiSq_All_vs_Cluster2$residuals


#2b. Test for enrichment of infected/uninfected cells in Cluster 2 versus the whole dataset
InfectedUninfectedCellCounts_All <- 
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    count(infected)

InfectedUninfectedCellCounts_Cluster1 <- 
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident == "0") %>%
    count(infected)

InfectedUninfectedCellCounts_Cluster2 <- 
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident == "2") %>%
    count(infected) 

#Build Contingency Tables
InfectedUninfected_ContingencyTable_All_vs_Cluster1 <- as.data.frame(matrix(nrow = 2, ncol = 2))
InfectedUninfected_ContingencyTable_All_vs_Cluster2 <- as.data.frame(matrix(nrow = 2, ncol = 2))

colnames(InfectedUninfected_ContingencyTable_All_vs_Cluster1) <- InfectedUninfectedCellCounts_All$infected
rownames(InfectedUninfected_ContingencyTable_All_vs_Cluster1) <- c("All", "Cluster1")
InfectedUninfected_ContingencyTable_All_vs_Cluster1[1,] <- InfectedUninfectedCellCounts_All$n
InfectedUninfected_ContingencyTable_All_vs_Cluster1[2,] <- InfectedUninfectedCellCounts_Cluster1$n

colnames(InfectedUninfected_ContingencyTable_All_vs_Cluster2) <- InfectedUninfectedCellCounts_All$infected
rownames(InfectedUninfected_ContingencyTable_All_vs_Cluster2) <- c("All", "Cluster2")
InfectedUninfected_ContingencyTable_All_vs_Cluster2[1,] <- InfectedUninfectedCellCounts_All$n
InfectedUninfected_ContingencyTable_All_vs_Cluster2[2,] <- InfectedUninfectedCellCounts_Cluster2$n

#Chi.Sq Testing
ChiSq_InfUninf_All_vs_Cluster1 <- chisq.test(x = InfectedUninfected_ContingencyTable_All_vs_Cluster1, simulate.p.value = F)
ChiSq_InfUninf_All_vs_Cluster2 <- chisq.test(x = InfectedUninfected_ContingencyTable_All_vs_Cluster2, simulate.p.value = F)

ChiSq_InfUninf_All_vs_Cluster1$residuals
ChiSq_InfUninf_All_vs_Cluster2$residuals





#Define Enrichment % of 2P4M cells in Cluster 2
PercentTha2P4MAstrocytes_in_All      <- as.numeric(round(ExperimentalCellCounts_All[4,2] / sum(ExperimentalCellCounts_All$n) * 100, digits = 3))
PercentTha2P4MAstrocytes_in_Cluster2 <- as.numeric(round(ExperimentalCellCounts_Cluster2[4,2] / sum(ExperimentalCellCounts_Cluster2$n) * 100, digits = 3))

Tha2P4MAstrocyteEnrichment_inCluster2 <- PercentTha2P4MAstrocytes_in_Cluster2 / PercentTha2P4MAstrocytes_in_All


#3. Viral Load Comparison across Cluster 1 and Cluster 2
tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster1 <-
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident == "0")

tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster2 <-
    tha.celltype.infections.summary_astrocytes_CLEAN2 %>%
    filter(ident == "2")

MeanViralLoad_Cluster1 <- round(mean(tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster1$ViralLoad * 100), digits = 1)
SEMViralLoad_Cluster1  <- round(stderror(tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster1$ViralLoad * 100), digits = 2)

MeanViralLoad_Cluster2 <- round(mean(tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster2$ViralLoad * 100), digits = 1)
SEMViralLoad_Cluster2  <- round(stderror(tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster2$ViralLoad * 100), digits = 2)

WilcoxTest_ViralLoad_Cluster1_vs_Cluster2 <- wilcox.test(x = tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster1$ViralLoad, 
                                                         y = tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster2$ViralLoad)





#4. DYNNL1 levels by viral load group across cell types
DYNNL1_Summary_CellTypeStratified_Neuron <-
  DYNNL1_Summary_CellTypeStratified %>%
  filter(Neuron_AstroLow_AstroHigh == "Neuron")

DYNNL1_Summary_CellTypeStratified_AstroLow <-
  DYNNL1_Summary_CellTypeStratified %>%
  filter(Neuron_AstroLow_AstroHigh == "AstroLow")

DYNNL1_Summary_CellTypeStratified_AstroHigh <-
  DYNNL1_Summary_CellTypeStratified %>%
  filter(Neuron_AstroLow_AstroHigh == "AstroHigh")

# One-way ANOVA testing
#Neuron
ks_DYNNL1_by_ViralLoadGroup_Neuron <- kruskal.test(DYNLL1 ~ ViralLoad_groups, data = DYNNL1_Summary_CellTypeStratified_Neuron)
summary(ks_DYNNL1_by_ViralLoadGroup_Neuron)

#Cluster 1 / AstroLow
ks_DYNNL1_by_ViralLoadGroup_Cluster1 <- kruskal.test(DYNLL1 ~ ViralLoad_groups, data = DYNNL1_Summary_CellTypeStratified_AstroLow)
summary(ks_DYNNL1_by_ViralLoadGroup_Cluster1)

#Cluster 2 / AstroHigh
ks_DYNNL1_by_ViralLoadGroup_Cluster2 <- kruskal.test(DYNLL1 ~ ViralLoad_groups, data = DYNNL1_Summary_CellTypeStratified_AstroHigh)
summary(ks_DYNNL1_by_ViralLoadGroup_Cluster2)



#5. ISG Score Testing
WilcoxTest_ISGScore_Cluster1_vs_Cluster2 <- wilcox.test(x = tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster1$ISGScore, 
                                                         y = tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster2$ISGScore)

WilcoxTest_IFNGenesAllScore_Cluster1_vs_Cluster2 <- wilcox.test(x = tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster1$IFNGenesAllScore, 
                                                         y = tha.celltype.infections.summary_astrocytes_CLEAN2_Cluster2$IFNGenesAllScore)

```
